src/libraries/System.DirectoryServices/src/System/DirectoryServices/ActiveDirectory/ActiveDirectorySchemaClass.cs

```diff
@@ -365,14 +365,14 @@ public void Save()
             try
             {
                 // commit the classEntry to server
-                _classEntry.CommitChanges();
+                _classEntry!.CommitChanges();
```

Comment:
That's technically correct if you construct this class in certain ways (not through public constructor but through FindByName).
You'd get ComException wrapping NRE though rather than pure NRE

The internal constructor which could cause this seems to be:
`internal ActiveDirectorySchemaClass(DirectoryContext context, string ldapDisplayName, DirectoryEntry? classEntry, DirectoryEntry? schemaEntry)`.

Likely constructor should do something like:
`_classEntry = classEntry ?? GetSchemaClassDirectoryEntry();`

or Save should throw (I'm not that familiar with AD to tell what the right answer is).

I'll leave this comment unresolved until I turn this into the issue.